test/language/module-code/top-level-await/dynamic-import-of-waiting-module.js

```diff
@@ -0,0 +1,69 @@
+  Dynamic import of an ~evaluating-async~ module waits for the module to finish its evaluation
+info: |
+  ContinueDynamicImport ( _promiseCapability_, _moduleCompletion_ )
+    1. ...
+    1. Let _module_ be _moduleCompletion_.[[Value]].
+    1. Let _loadPromise_ be _module_.LoadRequestedModules().
+    1. Let _rejectedClosure_ be a new Abstract Closure with parameters (_reason_) that captures _promiseCapability_ and performs the following steps when called:
+      1. Perform ! Call(_promiseCapability_.[[Reject]], *undefined*, « _reason_ »).
+      1. Return ~unused~.
+    1. Let _onRejected_ be CreateBuiltinFunction(_rejectedClosure_, 1, *""*, « »).
+    1. Let _linkAndEvaluateClosure_ be a new Abstract Closure with no parameters that captures _module_, _promiseCapability_, and _onRejected_ and performs the following steps when called:
+      1. Let _link_ be Completion(_module_.Link()).
+      1. ...
+      1. Let _evaluatePromise_ be _module_.Evaluate().
+      1. Let _fulfilledClosure_ be a new Abstract Closure with no parameters that captures _module_ and _promiseCapability_ and performs the following steps when called:
+        1. Let _namespace_ be GetModuleNamespace(_module_).
+        1. Perform ! <emu-meta effects="user-code">Call</emu-meta>(_promiseCapability_.[[Resolve]], *undefined*, « _namespace_ »).
+        1. Return ~unused~.
+      1. Let _onFulfilled_ be CreateBuiltinFunction(_fulfilledClosure_, 0, *""*, « »).
+      1. Perform PerformPromiseThen(_evaluatePromise_, _onFulfilled_, _onRejected_).
+      1. Return ~unused~.
+    1. Let _linkAndEvaluate_ be CreateBuiltinFunction(_linkAndEvaluateClosure_, 0, *""*, « »).
+    1. Perform PerformPromiseThen(_loadPromise_, _linkAndEvaluate_, _onRejected_).
+    1. Return ~unused~.
+
+  _module_ . Evaluate (  )
+    4. If _module_.[[TopLevelCapability]] is not ~empty~, then
+       a. Return _module_.[[TopLevelCapability]].[[Promise]].
+
+flags: [async]
+features: [dynamic-import]
+---*/
+
+let continueExecution;
+globalThis.promise = new Promise(res => continueExecution = res);
+
+const executionStartPromise = new Promise(res => globalThis.executionStarted = res);
+
+const promiseForNamespace = import("./dynamic-import-of-waiting-module_FIXTURE.js");
+
+executionStartPromise.then(() => {
+  const promiseForNamespace2 = import("./dynamic-import-of-waiting-module_FIXTURE.js");
+
+  // We only continuye execution of the first fixture file after importing a second,
+  // empty, fixture file. This is so that if the implementation uses a separate
+  // queue to resolve dynamic import promises, if dynamic-import-of-waiting-module_FIXTURE
+  // wasn't waiting on top-level await its top-level promise would already be resolved.
+  import("./dynamic-import-of-waiting-module-2_FIXTURE.js").then(() => {
+    continueExecution();
+  }, $DONE);
+
+  let secondPromiseResolved = false;
+  return Promise.all([
+    promiseForNamespace.then(() => {
+      console.log("Resolving first...");
```

Comment:
Remove, this won't go anywhere and I believe runners are not required to define `console`